src/ui/components/pages/swap/channelClient.ts

```diff
@@ -29,34 +34,43 @@ interface ExchangeRequest extends ExchangeInput {
   id: string;
 }
 
-export enum ExchangeChannelErrorType {
-  ConnectionError = 'CONNECTION_ERROR',
-  ExchangeError = 'EXCHANGE_ERROR',
-}
+class ExchangeChannelConnectionError {}
 
-export enum ExchangeChannelErrorCode {
-  INVALID_ASSET_PAIR = 'INVALID_ASSET_PAIR',
-  UNEXPECTED_ERROR = 'UNEXPECTED_ERROR',
-}
-
-export class ExchangeChannelError {
-  type: ExchangeChannelErrorType;
-  code: string;
+type Subscriber = (
+  err: ExchangeChannelConnectionError | null,
+  vendor?: string,
+  response?: ExchangeChannelResult
+) => void;
 
-  constructor(
-    type: ExchangeChannelErrorType,
-    code = ExchangeChannelErrorCode.UNEXPECTED_ERROR
-  ) {
-    this.type = type;
-    this.code = code;
+function convertArg(arg: proto.Response.Exchange.Argument): CallArg {
+  if ('integerValue' in arg) {
```

Comment:
It is better to check which field is filled.
This should work but i could be wrong :)

```javascript
switch (arg.value){
  case "integerValue":
     // ...
  case "binaryValue":
     // ...
  // ...
}```